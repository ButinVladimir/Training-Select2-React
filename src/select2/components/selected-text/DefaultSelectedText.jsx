import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

export default function DefaultSelectedText({
  value,
  onClick,
}) {
  return (
    <input type="text" className="select2-selected-text" readOnly value={value} onClick={onClick} />
  );
}

DefaultSelectedText.propTypes = {
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
