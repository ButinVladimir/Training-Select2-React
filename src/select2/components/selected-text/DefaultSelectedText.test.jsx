import React from 'react';
import renderer from 'react-test-renderer';
import DefaultSelectedText from './DefaultSelectedText';

describe('DefaultSelectedText', () => {
  test('renders without crashing', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultSelectedText value="value" onClick={mockFn} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).not.toHaveBeenCalled();
  });

  test('fires onClick properly', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultSelectedText value="value" onClick={mockFn} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).toHaveBeenCalled();
  });
});
