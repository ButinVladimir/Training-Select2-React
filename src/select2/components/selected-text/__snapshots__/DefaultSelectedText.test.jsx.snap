// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultSelectedText fires onClick properly 1`] = `
<input
  className="select2-selected-text"
  onClick={[Function]}
  readOnly={true}
  type="text"
  value="value"
/>
`;

exports[`DefaultSelectedText fires onClick properly 2`] = `
<input
  className="select2-selected-text"
  onClick={[Function]}
  readOnly={true}
  type="text"
  value="value"
/>
`;

exports[`DefaultSelectedText renders without crashing 1`] = `
<input
  className="select2-selected-text"
  onClick={[Function]}
  readOnly={true}
  type="text"
  value="value"
/>
`;
